bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/troubleshoot/StrandDump.java

```diff
@@ -18,59 +18,114 @@
+import com.sun.management.HotSpotDiagnosticMXBean;
 
+import javax.management.MBeanServer;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * Used to get the status of current Ballerina strands.
  *
  * @since 2201.2.0
  */
 public class StrandDump {
+    private static final String hotSpotBeanName = "com.sun.management:type=HotSpotDiagnostic";
+    private static volatile HotSpotDiagnosticMXBean hotSpotDiagnosticMXBean;
 
     public static String getStrandDump() {
-        Map<Integer, Strand> availableStrands = new HashMap<>();
-        int createdStrandGroupCount = 0;
-        int createdStrandCount = 0;
-        int availableStrandCount = 0;
-        Map<Integer, List<String>> availableStrandGroups = new HashMap<>();
-
-        String strandDumpOutput = generateOutput(availableStrandGroups, availableStrandCount, createdStrandGroupCount,
-                createdStrandCount);
-        cleanUp(availableStrands, availableStrandGroups);
-        return strandDumpOutput;
+        String filename = "thread_dump" + LocalDateTime.now();
+        String dump;
+        try {
+            getStrandDump(System.getProperty("user.dir") + "/" + filename);
+            dump = new String(Files.readAllBytes(Paths.get(filename)));
+            File fileObj = new File(filename);
+            fileObj.delete();
+        } catch (Exception e) {
+            return "Error occurred during strand dump generation";
+        }
+        return generateOutput(dump);
     }
 
-    private static String generateOutput(Map<Integer, List<String>> availableStrandGroups, int availableStrandCount,
-                                         int createdStrandGroupCount, int createdStrandCount) {
+    private static String generateOutput(String dump) {
+        String[] dump_items = dump.split("\\n\\n");
+        int id = 0;
+        Set<Integer> isolatedWorkerList = new HashSet<>();
+        Set<Integer> nonIsolatedWorkerList = new HashSet<>();
+        ArrayList<ArrayList<String>> balTraces = new ArrayList<>();
+        for(String item : dump_items) {
+            String[] lines = item.split("\\n");
+            String[] subitems = lines[0].split("\" ");
+            ArrayList<String> balTraceItems = new ArrayList<>();
+            boolean bal_strand = false;
+            if(subitems.length > 1 && subitems[1].equals("virtual")) {
+                balTraceItems.add("\tStrand " + lines[0].replace("virtual", ":") + "\n\t\tat");
+                String prefix = " ";
+                for(String line : lines) {
+                    if(line.contains("Scheduler.lambda$startIsolatedWorker")) {
+                        isolatedWorkerList.add(id);
```

Comment:
Can we continue if true

Reply:
Here we keep track of which particular strands are "isolated". But the conditions that come up next are responsible for adding this particular line to the stack trace if it meets the necessary criteria. Therefore, we will have to go through the checks that follow as well. I could make the check for "Non Isolated Workers" an else assuming a strand won't be both "Isolated" and "Non Isolated" at the same time.

Reply:
Discussed offline about ways of optimising the string parsing and restructuring logic.

(Fixed in d8ca5c11fb7c6a9a52e8a51f775f87eb2f451a56)